source/tutorial/enforce-keyfile-access-control-in-existing-sharded-cluster-no-downtime.txt

```diff
@@ -0,0 +1,297 @@
+your cluster only has one available :program:`mongos` instance, this results
+in downtime during the period that the :program:`mongos` is offline. To avoid
+this, ensure your sharded cluster has at least *two* :program:`mongos`
+instances available.
+
+.. _security-shardcluster-nodowntime-enable-access-control:
+
+Enforce Keyfile Access Control on an Existing Sharded Cluster
+-------------------------------------------------------------
+
+Create and Distribute the Keyfile
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. include:: /includes/extracts/keyfile-intro-sharded-cluster.rst
+
+.. code-block:: shell
+
+    openssl rand -base64 755 > <path-to-keyfile>
+    chmod 400 <path-to-keyfile>
+
+
+.. include:: /includes/extracts/keyfile-distribution-sharded-cluster.rst
+
+For more information on using keyfiles for internal authentication, refer to
+:ref:`internal-auth-keyfile`.
+
+.. _security-shardcluster-nodowntime-adminusers-clientusers:
+
+Configure Sharded Cluster Admin User and Client Users
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You must connect to a :program:`mongos` to complete these steps. The users
+created in these steps are cluster-level users and cannot be used for
+accessing individual shard replica sets.
+
+.. include:: /includes/steps/enable-authentication-in-shardcluster-nodowntime-uac.rst
+
+Restart Each ``mongos`` Instance with ``transitionToAuth`` set to ``true``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Repeat these steps for each :program:`mongos` in the sharded cluster,
+working on one :program:`mongos` at a time.
+
+.. include:: /includes/steps/enable-authentication-in-shardcluster-nodowntime-transition-mongos.rst
+
+At the end of this section, all :program:`mongos` instances in the sharded
+cluster are running with :setting:`security.transitionToAuth` and
+:setting:`security.keyFile` internal authentication.
+
+Restart Each Config Server Replica Set Member with ``transitionToAuth`` set to ``true``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Repeat these steps for each :program:`mongod` in the config server replica
+set, one member at a time. Work on the :term:`secondary` or :term:`arbiter`
+members *first*, and the :term:`primary` *last*.
+
+.. include:: /includes/steps/enable-authentication-in-shardcluster-nodowntime-transition-config.rst
+
+At the end of this section, all :program:`mongod` instances in the config
+server replica set is running with :setting:`security.transitionToAuth` and
+:setting:`security.keyFile` internal authentication.
+
+Restart Each Member of Each Shard Replica Set with ``transitionToAuth`` set to ``true``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Create the shard-local administrator
+++++++++++++++++++++++++++++++++++++
+
+Each shard replica set in a sharded cluster that enforces authentication
```

Comment:
In a sharded cluster that enforces authentication, each shard replica set should have its own shard-local administrator.